language_service/src/hover.rs

```diff
@@ -204,26 +300,68 @@ impl Visitor<'_> for HoverVisitor<'_> {
     }
 }
 
-fn markdown_with_doc(doc: &str, namespace: Option<Rc<str>>, code: impl Display) -> String {
-    let parsed_doc = parse_doc(doc);
+fn protocol_span(span: qsc::Span) -> protocol::Span {
+    protocol::Span {
+        start: span.lo,
+        end: span.hi,
+    }
+}
+
+fn curr_callable_to_params(curr_callable: Option<&ast::CallableDecl>) -> Vec<&ast::Pat> {
+    match curr_callable {
+        Some(decl) => vec![decl.input.as_ref()],
+        None => vec![],
+    }
+}
 
-    let code = match namespace {
-        Some(namespace) if !namespace.is_empty() => {
-            markdown_fenced_block(format!("{namespace}\n{code}"))
+fn is_param(param_pats: &[&ast::Pat], node_id: ast::NodeId) -> bool {
+    fn find_in_pat(pat: &ast::Pat, node_id: ast::NodeId) -> bool {
+        match &*pat.kind {
+            ast::PatKind::Bind(ident, _) => node_id == ident.id,
+            ast::PatKind::Discard(_) | ast::PatKind::Elided => false,
+            ast::PatKind::Paren(inner) => find_in_pat(inner, node_id),
+            ast::PatKind::Tuple(inner) => inner.iter().any(|x| find_in_pat(x, node_id)),
         }
-        _ => markdown_fenced_block(code),
-    };
+    }
 
-    if parsed_doc.summary.is_empty() {
-        code
-    } else {
-        format!("{}---\n{}\n", code, parsed_doc.summary,)
+    param_pats.iter().any(|pat| find_in_pat(pat, node_id))
+}
+
+fn display_local(
+    param_kind: &LocalKind,
+    markdown: &String,
+    local_name: &str,
+    callable_name: &str,
+    callable_doc: &str,
+) -> String {
+    match param_kind {
+        LocalKind::Param => {
+            let param_doc = parse_doc_for_param(callable_doc, local_name);
+            with_doc(
+                &param_doc,
+                format!("param of `{callable_name}`\n{markdown}",),
```

Comment:
"parameter of" would sound more human-friendly IMO 🙂 

Also, does the single line break `\n` work well visually? Markdown sometimes requires double line breaks for proper visual separation. (I can't tell from reading this code whether this will look OK, just worth checking)

Reply:
Changed to "parameter of". The single line break seems to work well. Let's follow up with that in the other comment.